controllers/vspheremachine_controller.go

```diff
@@ -108,7 +108,7 @@ func AddMachineControllerToManager(ctx context.Context, controllerManagerContext
 			).
 			Watches(
 				&clusterv1.Cluster{},
-				handler.EnqueueRequestsFromMapFunc(r.clusterToVMwareMachines),
+				handler.EnqueueRequestsFromMapFunc(r.enqueueClusterToMachineRequests),
```

Comment:
Sorry for stressing this and causing some more work, let's revert the changes around this for this PR and move them to a separate PR. (Includes reverting the additional new / moved functions)

I'm happy to help out if you don't find the time for it.